jax/_src/lax/control_flow.py

```diff
@@ -987,9 +987,11 @@ def _cond_partial_eval(trace, *tracers, branches, linear):
 
   linear_2 = (False,) * num_res + linear
   params = dict(branches=branches_2, linear=linear_2)
+  name_stack = source_info_util.current_name_stack()[len(trace.name_stack):]
```

Comment:
What's the logic to this pattern? I wonder if we could abstract it into a well-named function, since it's done a couple times.